apps/commudle-admin/src/app/services/hackathon.service.ts

```diff
@@ -428,4 +434,20 @@ export class HackathonService {
       params,
     });
   }
+
+  pGetUpcomingHackathon(when, limit?, after?): Observable<IPagination<IHackathon>> {
+    let params = new HttpParams().set('when', when);
+    if (limit) {
+      params = params.set('limit', limit);
+    }
+    if (after) {
+      params = params.set('after', after);
+    }
+    return this.http.get<IPagination<IHackathon>>(
+      this.apiRoutesService.getRoute(API_ROUTES.HACKATHONS.PUBLIC.UPCOMING_HACKATHONS),
+      {
+        params,
+      },
+    );
+  }
 }
```

Comment:
why this function named as upcoming hackathons, may be it will return past or all hackathons in future